File: adt2ext4/MACRO_BROWSER.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2026 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void MACRO_BROWSER (bool instrBrowser, bool updateCurInstr)
{
  tFIXED_SONGDATA *song = &songdata;

  DBG_ENTER ("MACRO_BROWSER");

  songdata_crc = Update32 (song, sizeof (*song), 0); // FIXME: why not CRC32_INITVAL?

  a2w_file_loader (false, !instrBrowser, true, false, updateCurInstr, NULL); // browse internal A2W data

  if (Update32 (song, sizeof (*song), 0) != songdata_crc) // FIXME: why not CRC32_INITVAL?
    module_archived = false;

  DBG_LEAVE (); //EXIT //MACRO_BROWSER
}

File: adt2ext4/pas/MACRO_BROWSER.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2026 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure MACRO_BROWSER(instrBrowser: Boolean; updateCurInstr: Boolean);
begin
  _dbg_enter ({$I %FILE%}, 'MACRO_BROWSER');

  songdata_crc := Update32(songdata,SizeOf(songdata),0);
  a2w_file_loader(FALSE,NOT instrBrowser,TRUE,FALSE,updateCurInstr); // browse internal A2W data
  If (Update32(songdata,SizeOf(songdata),0) <> songdata_crc) then
    module_archived := FALSE;

  _dbg_leave; //EXIT //MACRO_BROWSER
end;
